aten/src/ATen/native/mps/operations/Pooling.mm

```diff
@@ -319,93 +315,71 @@ static PoolSizes process_pool_sizes(const Tensor& input,
 
   int32_t leading_dims = input.dim() - pooling_dims;
 
-  at::Tensor t_input_size = intarrayref_to_tensor(input.sizes());
-  at::Tensor t_input_pooling_size = t_input_size.slice(/*dim=*/0, /*start=*/leading_dims);
+  std::vector<int64_t> kernel_size_expanded = copy_and_maybe_expand(kernel_size, pooling_dims);
+  std::vector<int64_t> stride_expanded = copy_and_maybe_expand(stride.empty() ? kernel_size : stride, pooling_dims);
+  std::vector<int64_t> padding_expanded = copy_and_maybe_expand(padding, pooling_dims);
+  std::vector<int64_t> dilation_expanded = copy_and_maybe_expand(dilation, pooling_dims);
 
-  at::Tensor t_kernel_size = intarrayref_to_tensor(kernel_size);
-  if (kernel_size.size() == 1) {
-    t_kernel_size.repeat(pooling_dims);
+  for (int64_t dim = 0; dim < pooling_dims; dim++) {
+    TORCH_CHECK(padding_expanded[dim] >= 0, op_name, ": pad must be non-negative");
+    TORCH_CHECK(padding_expanded[dim] * 2 <= kernel_size_expanded[dim],
+                op_name,
+                ": pad should be at most half of effective kernel size");
   }
 
-  at::Tensor t_stride = stride.empty() ? t_kernel_size.clone() : intarrayref_to_tensor(stride);
-  if (!stride.empty() && stride.size() == 1) {
-    t_stride.repeat(pooling_dims);
+  for (int64_t dim = leading_dims == 2 ? 1 : 0; dim < dims; dim++) {
```

Comment:
Same as above. Also, not sure if this is OK/faster/more readable to do so, but aren't `leading_dims == 2 ? 1 : 0` the same as `static_cast<int>(leading_dims == 2)`
```suggestion
  for (const auto  dim: c10::irange(leading_dims == 2 ? 1 : 0, dims)) {
```